src/python_extension.ts

```diff
@@ -20,14 +20,16 @@ export class PythonExtension {
 	private onChange(pythonApi : IExtensionApi) {
 		pythonApi.settings.onDidChangeExecutionDetails(
 			(resource: Resource) => {
-				log(`Active interpreter changed for: ${resource}`);
-				log(`Execution details: ${JSON.stringify(this.executionDetails)}`);
-				this.executionDetails = pythonApi.settings.getExecutionDetails(resource);
-				this.onChangeCallback();
+				let newExecutionDetails = pythonApi.settings.getExecutionDetails(resource);
+				if(this.executionDetails.execCommand[0] !== newExecutionDetails.execCommand[0]){
+					log(`Active interpreter changed for: ${resource}`);
+					log(`Execution details: ${JSON.stringify(this.executionDetails)}`);
+					this.onChangeCallback();
+				}
+				this.executionDetails = newExecutionDetails;
```

Comment:
The respective order of these two statements changed. I don't know what `onChangeCallback` does so I can't really estimate if this is a potential issue.

Reply:
I don't think it matters, but it's easy to use the same order so I made the change.